File: src/main/java/com/example/jobs/entity/Job.java
package com.example.jobs.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.jspecify.annotations.Nullable;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Setter
@Table(name = "jobs")
public class Job implements Persistable<UUID> {

  @Id
  private UUID id;

  @Transient
  private boolean isNewRecord = true;

  @Column("company_id")
  private Long companyId;

  @Column("title")
  private String title = "";

  @Column("slug")
  private String slug = "";

  @Column("description")
  private String description;

  @Column("salary_min")
  private BigDecimal salaryMin;

  @Column("salary_max")
  private BigDecimal salaryMax;

  @Column("location")
  private String location;

  @Column("workplace")
  private String workplace;

  @Column("job_type")
  private String jobType;

  @Column("experience_level")
  private String experienceLevel;

  @Column("apply_url")
  private String applyUrl;

  @Column("is_featured")
  private Boolean isFeatured = false;

  @Column("is_sticky")
  private Boolean isSticky = false;

  @Column("status")
  private String status;

  @Column("quality_score")
  private Integer qualityScore;

  @Column("url")
  private String url;

  @Column("published_at")
  private LocalDateTime publishedAt;

  @Column("expires_at")
  private LocalDateTime expiresAt;

  @Column("created_at")
  private LocalDateTime createdAt;

  @Column("updated_at")
  private LocalDateTime updatedAt;

  @Override
  public @Nullable UUID getId() {
    return null;
  }

  @Override
  public boolean isNew() {
    return false;
  }

  public void setNew(Boolean isNew) {
    this.isNewRecord = isNew;
  }

  public void setIsNewRecord(Boolean newRecord) {
    this.isNewRecord = newRecord;
  }
}

File: src/main/kotlin/com/example/jobs/entity/Job.kt
package com.example.jobs.entity

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

@Table(name = "jobs")
class Job : Persistable<UUID> {

    private var id: UUID? = null

    @Transient
    private var isNewRecord: Boolean = true

    @Column("company_id")
    var companyId: Long? = null

    @Column("title")
    var title: String = ""

    @Column("slug")
    var slug: String = ""

    @Column("description")
    var description: String? = null

    @Column("salary_min")
    var salaryMin: BigDecimal? = null

    @Column("salary_max")
    var salaryMax: BigDecimal? = null

    @Column("location")
    var location: String? = null

    @Column("workplace")
    var workplace: String? = null

    @Column("job_type")
    var jobType: String? = null

    @Column("experience_level")
    var experienceLevel: String? = null

    @Column("apply_url")
    var applyUrl: String? = null

    @Column("is_featured")
    var isFeatured: Boolean? = false

    @Column("is_sticky")
    var isSticky: Boolean? = false

    @Column("status")
    var status: String? = null

    @Column("quality_score")
    var qualityScore: Int? = null

    @Column("url")
    var url: String? = null

    @Column("published_at")
    var publishedAt: LocalDateTime? = null

    @Column("expires_at")
    var expiresAt: LocalDateTime? = null

    @Column("created_at")
    var createdAt: LocalDateTime? = null

    @Column("updated_at")
    var updatedAt: LocalDateTime? = null

    @Id
    override fun getId(): UUID? = this.id

    override fun isNew(): Boolean = this.isNewRecord

    fun setId(id: UUID) {
        this.id = id
    }

    // Update the correct consolidated flag
    fun setNew(isNew: Boolean) {
        this.isNewRecord = isNew
    }

    fun setIsNewRecord(newRecord: Boolean) {
        this.isNewRecord = newRecord
    }
}